### ISSUE TITLE
Improve Tile Inspector Delete

### ISSUE THREAD

**Issue by duncanspumpkin** [OpenRCT2/OpenRCT2#10691]
When deleting tiles with the tile inspector it shouldn't be possible to delete all the tiles at one coordinate. There should always be at least one (it could also be argued there should always be a surface element as many functions assume that).

**Comment by Broxzier** [comment]
> it could also be argued there should always be a surface element as many functions assume that

There are parks that have deleted surface elements to make space for scenery. Of course, these are extreme cases, but I'd argue that support for this should be kept since the game can already handle it.
On the other hand, if we can guarantee that surface elements always exist on every tile, many null checks can be removed.

**Comment by kylegarza21** [comment]
@duncanspumpkin am I free to give this one a go?

**Comment by kylegarza21** [comment]
@duncanspumpkin @Broxzier ,

I've identified a path forward and would like to validate my interpretation using the below snippet. The 'floating' tree should not be able to be deleted nor should other tiles that contain only one element such as the simple grass tiles

I'm thinking that a warning message would be in scope of this work due to restricting functionality. Please let me know your thoughts on this

![image](https://user-images.githubusercontent.com/28277652/76136772-465eae80-6003-11ea-9414-4aca56449afb.png)

**Comment by duncanspumpkin** [comment]
Yes that is correct. There should always be at least one tile doesn't matter what type. I strongly advise that we shouldn't delete surface elements either and I think maybe when we change save format that will be enforced as the only reason to delete a surface element is to save on element space.

**Comment by IntelOrca** [comment]
> I strongly advise that we shouldn't delete surface elements either.

Sure but you can't disallow it since some people still want to remove them.

**Comment by kylegarza21** [comment]
Please assign me to this issue; I started a [branch](https://github.com/OpenRCT2/OpenRCT2/compare/develop...kylegarza21:fix/10691?expand=1)

**Comment by duncanspumpkin** [comment]
@kylegarza21 there is no way to assign people who aren't team members. Just submit a PR when you have something.
